Assets/Scripts/Managers/BattleQueue.cs

```diff
@@ -280,11 +410,426 @@ public List<ActionClass> GetList()
+                    {
+                        Swap(wrappers, y - 1, y);
+                    }
+                    y--;
+                }
+            }
+        }
+
+
+
+
+        public List<Wrapper> GetWrappers()
+        {
+            return wrappers;
+        }
+
+    }
+
+    // Wrapper Element for WrapperArray;
+    public class Wrapper
+    {
+        public ActionClass PlayerAction { get; set; }
+        public ActionClass EnemyAction { get; set; }
+
+        public int HighestSpeed { get; set; } // used to sort the wrappers 
+                                                // -1 indicates that the wrapper is empty 
+                                                // bad encapsulation as maintenance is done outside.
+
+        // Every enemy action is transformed into a single field wrapper after all the enemy actions have been inserted;
+        // Will implement this via a field checker (roundStart) at present; should be implemented elsewhere for decorum. 
+
+        // the true corresponds to enemy action; false to player action
+        // TO REFACTOR !!! isEnemys
+        public Wrapper(ActionClass action, bool isEnemys)
+        {
+            if (isEnemys)
+            {
+                PlayerAction = null;
+                this.EnemyAction = action;
+                HighestSpeed = action.Speed;
+            }
+            else
+            {
+                this.PlayerAction = action;
+                EnemyAction = null;
+                HighestSpeed = action.Speed;
+            }
+        }
+
+        // returns the action with the highest speed. 
+        // IS NOT RESPONSIBLE FOR THE DESTRUCTION AS THE DESTRUCTION MUST TAKE PLACE AFTER THE EXECUTION; for destruction check speed. 
+        // call in dequeue. 
+        // updates Highest Speed. 
+        public ActionClass ReturnHighest()
+        {
+            if (PlayerAction == null) // account for half wrapper. 
+            {
+                HighestSpeed = -1;
+                ActionClass temp = EnemyAction;
+                EnemyAction = null;
+                return temp;
+            }
+            else if (EnemyAction == null)
+            {
+                HighestSpeed = -1;
+                ActionClass temp = PlayerAction;
+                PlayerAction = null;
+                return temp;
+            }
+            else
+            {
+                if (PlayerAction.Speed >= EnemyAction.Speed)
+                {
+                    HighestSpeed = EnemyAction.Speed;
+                    ActionClass temp = PlayerAction;
+                    PlayerAction = null;
+                    return temp;
+                }
+                else
+                {
+                    HighestSpeed = PlayerAction.Speed;
+                    ActionClass temp = EnemyAction;
+                    EnemyAction = null;
+                    return temp;
+                }
+            }
+        }
```

Comment:
Why does this unassign the field that was the highest speed? This side effect could potentially be dangerous?